### PR TITLE
 refactor: private hooks directory

### PR DESCRIPTION
**Pull request by NikitaCG**

### CODE REVIEW

File: src/hooks/private/useCheckbox/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# useCheckbox
+
+The `useCheckbox` hook need to generate props for radio group element
```
Comment (amje) [inline src/hooks/private/useCheckbox/README.md:3]: radio group element -> checkbox control

File: src/hooks/private/useCheckbox/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+| onUpdate       | OnUpdate callback           |                                                                `(checked: boolean) => void`                                                                |         |
+| onChange       | OnChange callback           |                                                            `(event: React.ChangeEvent) => void`                                                            |         |
+| onFocus        | OnFocus callback            |                                                            `(event: React.FocusEvent) => void`                                                             |         |
+| onBlur         | OnBlur callback             |                                                            `(event: React.FocusEvent) => void`                                                             |         |
+| id             | ID attribute                |                                                            `(event: React.FocusEvent) => void`                                                             |         |
+
+## Result
+
+- checked, radio-input status `boolean`
```
Comment (amje) [inline src/hooks/private/useCheckbox/README.md:25]: Let's use md table here, checked - Checked state, inputProps - Props to pass to the input element

File: src/hooks/private/useCloseOnTimeout/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+| Name    | Description       |      Type      | Default |
+| :------ | :---------------- | :------------: | :-----: |
+| onClose | On close callback | `VoidFunction` |         |
+| timeout | Timeout           |    `number`    |         |
+
+## Result
+
+- onMouseOver, `React.MouseEventHandler`
+- onMouseLeave, `React.MouseEventHandler`
```
Comment (amje) [inline src/hooks/private/useCloseOnTimeout/README.md:15]: md table

File: src/hooks/private/useRadio/README.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# useRadio
+
+The `useRadio` hook need to generate props for radio group element
```
Comment (amje) [inline src/hooks/private/useRadio/README.md:3]: element -> control

File: src/hooks/private/useRadio/README.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+| onUpdate       | OnUpdate callback           |                                                                `(checked: boolean) => void`                                                                |         |
+| onChange       | OnChange callback           |                                                            `(event: React.ChangeEvent) => void`                                                            |         |
+| onFocus        | OnFocus callback            |                                                            `(event: React.FocusEvent) => void`                                                             |         |
+| onBlur         | OnBlur callback             |                                                            `(event: React.FocusEvent) => void`                                                             |         |
+| id             | ID attribute                |                                                            `(event: React.FocusEvent) => void`                                                             |         |
+
+## Result
+
+- checked, radio-input status `boolean`
+- inputProps
```
Comment (amje) [inline src/hooks/private/useRadio/README.md:25]: Same as for useCheckbox

File: src/hooks/private/useRadioGroup/README.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# useRadioGroup
+
+The `useRadioGroup` hook need to generate props for radio group element
```
Comment (amje) [inline src/hooks/private/useRadioGroup/README.md:3]: element -> control

File: src/hooks/private/useRadioGroup/README.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+| disabled     | Disabled flag     |                                           `boolean`                                            |         |
+| onUpdate     | OnUpdate callback |                                   `(value: string) => void`                                    |         |
+| onChange     | OnChange callback |                              `(event: React.ChangeEvent) => void`                              |         |
+| onFocus      | OnFocus callback  |                              `(event: React.FocusEvent) => void`                               |         |
+| onBlur       | OnBlur callback   |                              `(event: React.FocusEvent) => void`                               |         |
+
+## Result
+
+- containerProps
+- optionsProps
```
Comment (amje) [inline src/hooks/private/useRadioGroup/README.md:22]: Same as for useCheckbox

File: src/hooks/private/useRestoreFocus/README.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+| Name            | Description                |       Type        | Default |
+| :-------------- | :------------------------- | :---------------: | :-----: |
+| enabled         | Enabled flag               |     `boolean`     |         |
+| restoreFocusRef | Ref-link for restore focus | `React.RefObject` |         |
+| focusTrapped    | Focus trapped flag         |     `boolean`     |         |
+
+## Result
+
+- onFocus, `(event: React.FocusEvent) => void`
```
Comment (amje) [inline src/hooks/private/useRestoreFocus/README.md:15]: md table

File: src/hooks/private/useStateWithCallback/README.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+
+| Name         | Description          |          Type          | Default |
+| :----------- | :------------------- | :--------------------: | :-----: |
+| initialState | initial state        |         `any`          |         |
+| callback     | callback for wrapper | `(value: any) => void` |         |
+
+## Result
+
+- state, `any`
+- setWithCallback, `(nextValue: T | ((value: T) => T)) => void`
```
Comment (amje) [inline src/hooks/private/useStateWithCallback/README.md:15]: md table

**Review by amje — CHANGES_REQUESTED** [review]

**Review by amje — APPROVED** [review]

**Review by korvin89 — APPROVED** [review]

File: src/hooks/private/index.ts [hunk 10]
```diff
@@ -0,0 +1,11 @@
+export * from './useBoolean';
+export * from './useCheckbox';
+export * from './useCloseOnTimeout';
+export * from './useConditionallyControlledState';
```
Comment (kseniya57) [inline src/hooks/private/index.ts:4]: Why is this private hook? It's general, it can be used by any app.

Comment (NikitaCG) [reply src/hooks/private/index.ts:4]: it was not exported outside

Comment (kseniya57) [reply src/hooks/private/index.ts:4]: Ok ✅ 
Seems like it will be good to make it public.

**Review by kseniya57 — APPROVED** [review]

**Pull request merged**